chromoscores/snipping.py

```diff
@@ -1,47 +1,88 @@
 import numpy as np
 
 
-def peak_snipping(contact_map, window_size, peak_coordinate):
-    """
-    Peak snippet; snippet with a "window_size" around peak with "peak_coordinate"
-    on a contact map, "contact_map". peak_coordinate should be in the format
-     of (i,j)
-    -----------------
-    Function peak_snippet(contact_map, window_size, peak_coordinate):
-
-    begin function
-
-         raise an error if peak coordinate + window window_size is out of the map
-
-         snippet = contact_map[
-        (peak_coordinate[0] - window_size):(peak_coordinate[0] + window_size),
-        (peak_coordinate[1] - window_size):(peak_coordinate[1] + window_size)]
 
-     return snippet_matrix
+def get_diagonal_pile_up(contact_map, boundary_list, window_size):
+    """
+    parameters
+    ----------
+    contact_map: contact map
+    boundary_list: list of the boundary elements positions on the diagonal 
+    window_size: size of the window
+
+    Returns
+    -------
+    a stackup of snippts around the boundary elements
+    """
+    mat = np.zeros((window_size, window_size))
+    for i in range(len(boundary_list)):
+        mat += contact_map[
+            boundary_list[i] - window_size // 2 : boundary_list[i] + window_size // 2,
+            boundary_list[i] - window_size // 2 : boundary_list[i] + window_size // 2,
+        ]
+    return mat
+
+def get_expected_map(contact_map):
```

Comment:
move to maputils.py, rename to `get_observed_over_expected()` and add a note to compare with cooltools implementation https://github.com/open2c/cooltools/blob/2ec2a8ae482b39967f858ffe6ce231e6c0dfec8b/cooltools/lib/numutils.py#L564